### PR TITLE
[Doc] Add Python docstring in several places

### PR DESCRIPTION
**Pull request by strongoier**
Related issue = #2579

### CODE REVIEW

File: python/taichi/lang/__init__.py [hunk 1]
```diff
@@ -77,6 +77,15 @@ def memory_profiler_print():
 
 extension = _ti_core.Extension
 is_extension_supported = _ti_core.is_extension_supported
+"""Checks whether an extension is supported on an arch.
```
Comment (k-ye) [inline python/taichi/lang/__init__.py:80]: Let's use `def is_extension_supported(arch, ext)`...

File: python/taichi/lang/type_factory_impl.py [hunk 2]
```diff
@@ -3,6 +3,10 @@
 
 
 class TypeFactory:
+    """Custom type factory for quantization.
```
Comment (k-ye) [inline python/taichi/lang/type_factory_impl.py:6]: ditto

File: python/taichi/lang/quant_impl.py [hunk 3]
```diff
@@ -5,6 +5,10 @@
 
 
 class Quant:
+    """Generator of quantized types.
```
Comment (k-ye) [inline python/taichi/lang/quant_impl.py:8]: Could you also document the methods within this class?

File: python/taichi/lang/__init__.py (hunk 1 above)
Comment (strongoier) [reply python/taichi/lang/__init__.py]: done

File: python/taichi/lang/type_factory_impl.py (hunk 2 above)
Comment (strongoier) [reply python/taichi/lang/type_factory_impl.py]: done

File: python/taichi/lang/quant_impl.py (hunk 3 above)
Comment (strongoier) [reply python/taichi/lang/quant_impl.py:8]: done

**Review by k-ye — APPROVED** [review]
LGTM~

**Pull request merged**